### PR TITLE
Feature/hal pwm change duty units

### PR DESCRIPTION
**Pull request by RobertoAldea**
Update, m% deleted now the setDuty works with values between min duty and HAL_PWM_period -1

### CODE REVIEW

File: firmware/Sources/Test/HAL/hal_pwm_test.c [hunk 1]
```diff
@@ -58,7 +63,7 @@
 HAL_PWM_result_e HAL_PwmTest(void)
 {
 	HAL_PWM_result_e res = HAL_PWM_RESULT_ERROR;
-	uint32_t duty= 50000; // Duty at 50%
+	uint32_t duty= 0.5 * HAL_PWM_period; // Duty at 50%
```
Comment (pastorpflores) [inline firmware/Sources/Test/HAL/hal_pwm_test.c:66]: More elegant : HAL_PWM_period / 2

File: firmware/Sources/Test/HAL/hal_pwm_test.c [hunk 2]
```diff
@@ -67,13 +72,13 @@ HAL_PWM_result_e HAL_PwmTest(void)
 			res = HAL_PwmStop();
 			if (res == HAL_PWM_RESULT_SUCCESS){
 				HAL_Delay(2500);
-				duty = 75000; // Duty 75%
+				duty =  0.75 * HAL_PWM_period; // Duty 75%
```
Comment (pastorpflores) [inline firmware/Sources/Test/HAL/hal_pwm_test.c:75]: HAL_PWM_period * 3 / 4

File: firmware/Sources/Test/HAL/hal_pwm_test.c [hunk 3]
```diff
@@ -67,13 +72,13 @@ HAL_PWM_result_e HAL_PwmTest(void)
...
 				res = HAL_PwmSetDuty(duty);
 				if (res == HAL_PWM_RESULT_SUCCESS){
 					res =HAL_PwmStart();
 					HAL_Delay(2500);
 					if (res == HAL_PWM_RESULT_SUCCESS){
-						duty = 25000; // Duty 25%
+						duty =  0.25 * HAL_PWM_period; // Duty 25%
```
Comment (pastorpflores) [inline firmware/Sources/Test/HAL/hal_pwm_test.c:81]: HAL_PWM_period / 4

**Review by pastorpflores — CHANGES_REQUESTED** [review]

File: firmware/Sources/Test/HAL/hal_pwm_test.c (hunk 2 above)
Comment (RobertoAldea) [reply firmware/Sources/Test/HAL/hal_pwm_test.c]: Solved

File: firmware/Sources/Test/HAL/hal_pwm_test.c (hunk 3 above)
Comment (RobertoAldea) [reply firmware/Sources/Test/HAL/hal_pwm_test.c]: Solved

File: firmware/Sources/Test/HAL/hal_pwm_test.c (hunk 1 above)
Comment (RobertoAldea) [reply firmware/Sources/Test/HAL/hal_pwm_test.c]: Solved

**Review by pastorpflores — APPROVED** [review]

**Pull request merged**